2023/day20.md

```diff
@@ -1,10 +1,153 @@
-- Run ZAP proxy (from installed location, in Mint it is by default in /opt/zaproxy)
-- Set env variables for ZAP_API_KEY and ZAP_PORT
-- Run several commands with zap cli. For example: zap-cli quick-scan -s all --ajax-spider -r http://127.0.0.1:8080/WebGoat/login.mvc . You should see some results in contrast UI.
+   IAST solution from here - <https://github.com/rstatsinger/contrast-java-webgoat-docker>. You need docker and docker-compose installed in mac or linux enviroment (this lab is tested on Mint). Please follow the README to create account in Contrast.
+
+## Getting started
+
+To start, clone the [repository](https://github.com/rstatsinger/contrast-java-webgoat-docker).
+
+Get your credentials from Contrast Security.
+Click on your name in the top-right corner -> `Organization Settings` -> `Agent`.
+Get the values for `Agent Username`, `Agent Service Key` and `API Key`.
+
+Replace these values in the `.env.template` file in the newly cloned repository.
+
+**NOTE:** These values are secret.
+Do not commit them to Git.
+It's best to put the `.env.template` under `.gitignore` so that you don't commit these values by mistake.
+
+## Running the vulnerable application
+
+To run the vulnerable application, run:
+
+```sh
+./run.sh
+```
+
+or
+
+```sh
+docker compose up
+```
+
+Once ready, the application UI will be accessible on <http://localhost:8080/WebGoat>.
+
+## Do some damage
+
+Now that we have a vulnerable application let's try to exploit it.
+
+1. Install ZAP Proxy from [here](https://www.zaproxy.org/download/)
+
+   An easy way to do that is via a DAST scanner.
+   One such scanner is [ZAP Proxy](https://www.zaproxy.org/).
+   It is a free and open-source web app scanner.
+
+2. Install `zap-cli` from [here](https://github.com/Grunny/zap-cli)
+
+   Next, install `zap-cli`.
+   `zap-cli` is an open-source CLI for ZAP Proxy.
+
+3. Run ZAP proxy
+
+   Run ZAP Proxy from its installed location.
+   In Linux Mint it is by default in `/opt/zaproxy`.
+   In MacOS it is in `Applications`.
+
+4. Set env variables for `ZAP_API_KEY` and `ZAP_PORT`
+
+   Get these values from ZAP Proxy.
+   Go to `Options...` -> `API` to get the API Key.
+
+   Go to `Options...` -> `Network` -> `Local Servers/Proxies` to configure and obtain the port.
+
+5. Run several commands with `zap-cli`
+
+   For example:
+
+   ```sh
+   zap-cli quick-scan -s all --ajax-spider -r http://127.0.0.1:8080/WebGoat/login.mvc
+   ```
+
+   Alternatively, you can follow the instructions in the [repo](https://github.com/rstatsinger/contrast-java-webgoat-docker/blob/master/Lab-WebGoat.pdf)
+   to cause some damage to the vulnerable application.
+
+6. Observe findings in Constrast
+
+   Either way, if you go to the **Vulnerabilities** tab for your application in Contrast you should be able to see that Contrast detected the vulnerabilities
+   and is warning you to take some action.
+
+## Bonus: Image Scanning
+
+We saw how an IAST solution helped us detect attacks in real time.
```

Comment:
In this case it is not in real time. Real time implementation do the similar job, mostly in the same manner, but vendors called them RASP (real-time application security protection). They are capable of blocking requests (the same way WAF does it) and IAST is for detection only. RASP is implemented in operations phase of the SDLC, not during development